packages/gasket-plugin-elastic-apm/lib/index.js

```diff
@@ -29,20 +32,53 @@ module.exports = {
   hooks: {
     configure: {
       handler: async (gasket, config) => {
+        const { logger } = gasket;
         config.elasticAPM = config.elasticAPM || {};
+
+        const { serverUrl, secretToken } = config.elasticAPM;
+        if (isDefined(serverUrl)) {
+          logger.notice('DEPRECATED config `elasticAPM.serverUrl`. Use env var: ELASTIC_APM_SERVER_URL');
+        }
+        if (isDefined(secretToken)) {
+          logger.notice('DEPRECATED config `elasticAPM.secretToken`. Use env var: ELASTIC_APM_SECRET_TOKEN');
+        }
+
         // eslint-disable-next-line no-process-env
         config.elasticAPM.active = isActive(config.elasticAPM, process.env);
 
         return { ...config };
       }
     },
     preboot: {
-      handler: async ({ config }) => {
-        require('elastic-apm-node')
-          .start({
+      handler: async (gasket) => {
+        const { config, logger } = gasket;
+
+        // prefer app-level dependency in case of duplicates
+        const apm = require(
+          require.resolve('elastic-apm-node', { paths: [config.root, __dirname] })
```

Comment:
Is this a use case for `peerDependencies` instead?

Reply:
That is correct. Switching the plugin to dev+peer dependencies would be a breaking change for current users of the plugin. So, we'll add it to new apps during create, but leave as a plugin dep for now until the next major release.